Discus Multiple Scattering Usability Enhancements

DannyHindson opened the pull request:
**Description of work.**

This is a slightly miscellaneous set of improvement\bug fixes to the inelastic version of the Discus multiple scattering correction. These are from feedback and other issues I've noticed since this sizeable change went in a few weeks ago as part of PR https://github.com/mantidproject/mantid/pull/33609:

- fix validation on ScatteringCrossSection parameter
- remove implicit assumption that w bins in S(Q,w) all have to be same size
- include results workspace group name in the member workspace names. If user runs any of the output workspaces through any further Mantid algorithms it's easier to see which workspace group they come from
- add some extra output workspaces (an extra "summed_1_n" workspace and Integrated workspaces where weights have been integrated across the x axis)
- Fix bug where vmu variable wasn't being updated after each scatter. Only affects inelastic calculations - effect on results is v small
- Remove a "2" from the equations in the documentation page. This shouldn't there and wasn't in the code
- Update Absorption\Multiple Scattering concepts page - this still said the algorithm was elastic only

**To test:**

Run the script on https://github.com/mantidproject/mantid/pull/33609
Check it completes successfully and also that the workspaces inside the results workspace group have names that all begin "MuscatResults..."

*There is no associated issue.*

<!-- Ensure the base of this PR is correct (e.g. release-next or master)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DannyHindson committed:
Remove spurious additional factor of 2

Remove from documentation of importance samplings
The factor of 2 isn't in the code and shouldn't be there. I think this
was put in to support having a factor of 2 in the denominator (2k^2)
but you don't need to force this. There's a factor of 2 in the denominator
naturally once the integration over phi is done

DannyHindson committed:
Add group name to the member workspace names

Add the group name as a prefix to the workspaces that are members of the
results workspace group
If you want to apply any algorithms to the result workspaces (eg Integrate
or Sum) and you have multiple results workspace groups this makes it
easier to work out which results set you're operating on
Update doc usage examples to accommodate this change and also fix up
some text that should have been changed when inelastic support introduced

DannyHindson committed:
Add in additional workspaces to provide ratio

New workspaces provide ratio of "single scatter" weight to "all scatter" weight.
To be used when applying correction using ratio method

DannyHindson committed:
Add integration workspaces for each scatter order

As requested by Spencer. Rather than call out to Integration algorithm
stick with the internal integration logic. Seems more consistent and
also perhaps more reliable on point datasets
Tidy up the integrateCumulative method putting interpolation logic into
named lambda and also supporting integration on bin edges as well as points
A few other misc changes:
- Correct unit validation on sigma_k input
- Remove spurious space from summed workspace names
- Small change to interpolateSquareRoot because x() was showing up in
performance profiler
- Remove implicit assumption that w bins were equal size - didn't need
to make this assumption

DannyHindson committed:
Fix bug where mu wasn't being updated after each scatter

The mu variable (density x cross section) wasn't being updated after
each scatter. This fix will have a small impact on inelastic calculations

DannyHindson committed:
Update a couple of comments

DannyHindson committed:
Adjust workflow algorithm tests to expect Integrated workspaces

DannyHindson committed:
Resolve compiler warnings

DannyHindson committed:
Add another output workspace containing the ratio of All to 1

Requested by Spencer for use in a ConvFit process for inelastic
Also add release note for fixes\new features that affect elastic
calculation (which was present in 6.3 unlike the inelastic calc)

DannyHindson committed:
Adjust test to accommodate second ratio workspace

DannyHindson committed:
Add detector info into Integrated workspaces

DannyHindson committed:
Adjust tests now that integrated ws have units on axes

robertapplin reviewed (changes requested):
The script runs successfully and the code looks good. I just had a few picky comments/suggestions, feel free to ~~`emplace_back`~~ push back on some of them

DannyHindson commented on Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp:
```diff
@@ -778,46 +814,72 @@ void DiscusMultipleScatteringCorrection::integrateCumulative(const Mantid::Histo
 }
 
+API::MatrixWorkspace_sptr DiscusMultipleScatteringCorrection::integrateWS(const API::MatrixWorkspace_sptr &ws) {
+  auto retVal = DataObjects::create<Workspace2D>(*ws, HistogramData::Points{0.});
```
It's a workspace containing an integral for each spectrum. I'll change the variable name to wsIntegrals

DannyHindson committed:
Incorporate review comments

Add some const and & qualifiers
Reserve memory for edges vector ahead of calling convertToBinBoundary
Improve linearInterp lambda
Use ternary operator instead of if statement

robertapplin reviewed (approved):
Thanks for the changes, looks good to me!
